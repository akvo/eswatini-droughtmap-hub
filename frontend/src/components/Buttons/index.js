export { default as LogoutButton } from "./LogoutButton";
export { default as SubmitButton } from "./SubmitButton";
export { default as SubmitReviewButton } from "./SubmitReview";
