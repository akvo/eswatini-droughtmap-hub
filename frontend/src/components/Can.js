"use client";

import { useUserContext } from "@/context/UserContextProvider";
import { defineUserAbility } from "@/lib";

const Can = ({ I, children, a = null, an = null }) => {
  const subject = a || an;
  const userContext = useUserContext();
  const userAbilities = userContext?.abilities || [];
  const ability = defineUserAbility(userAbilities);
  return ability.can(I, subject, "owner") ? children : null;
};

export default Can;
