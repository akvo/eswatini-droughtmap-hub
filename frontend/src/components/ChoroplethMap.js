"use client";

import { DEFAULT_CENTER } from "@/static/config";
import { Map } from "./Map";

const ChoroplethMap = ({ geoData }) => {
  const mapPolygonColorToDensity = (median) => {
    console.log("median", median);
    if (!median) {
      return "transparent";
    }
    return median <= 2
      ? "#730000"
      : median > 2 && median <= 5
      ? "#E60000"
      : median > 5 && median <= 10
      ? "#FFAA00"
      : median > 10 && median <= 20
      ? "#FCD37F"
      : median > 20 && median <= 30
      ? "#FFFF00"
      : "transparent";
  };
  const style = (feature) => {
    return {
      fillColor: mapPolygonColorToDensity(feature.properties.median),
      weight: 1,
      opacity: 1,
      color: "white",
      dashArray: "2",
      fillOpacity: 1,
    };
  };

  return (
    <Map className="w-full h-screen" center={DEFAULT_CENTER} zoom={12}>
      {({ TileLayer, Marker, Popup, GeoJSON }) => (
        <>
          <TileLayer
            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
            attribution='&copy; <a href="http://osm.org/copyright">OpenStreetMap</a> contributors'
          />
          <Marker position={DEFAULT_CENTER}>
            <Popup>Mbabane, Eswatini</Popup>
          </Marker>
          <GeoJSON data={geoData} style={style} />
        </>
      )}
    </Map>
  );
};

export default ChoroplethMap;
