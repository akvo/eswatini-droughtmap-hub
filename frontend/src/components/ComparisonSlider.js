"use client";

import { ReactCompareSlider } from "react-compare-slider";

const ComparisonSlider = ({ searchParams }) => {
  return (
    <div className="w-full">
      <ReactCompareSlider
        itemOne={
          <iframe
            width={"100%"}
            height={720}
            src={`/iframe/map?id=${searchParams?.left || 0}`}
          />
        }
        itemTwo={
          <iframe
            width={"100%"}
            height={720}
            src={`/iframe/map?id=${searchParams?.right || 1}`}
          />
        }
        boundsPadding={0}
        clip="both"
        keyboardIncrement="5%"
        position={50}
      />
    </div>
  );
};

export default ComparisonSlider;
