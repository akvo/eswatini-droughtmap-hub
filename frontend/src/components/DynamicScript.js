"use client";

import Script from "next/script";
import { feature } from "topojson-client";
import { useAppDispatch } from "@/context/AppContextProvider";

const DynamicScript = () => {
  const appDispatch = useAppDispatch();
  return (
    <Script
      src="/config.js"
      onLoad={() => {
        if (window?.topojson) {
          const geoData = feature(
            window.topojson,
            window.topojson.objects[Object.keys(window.topojson.objects)[0]]
          );
          appDispatch({
            type: "SET_GEODATA",
            payload: geoData,
          });

          const { features } = geoData;
          const administrations = features
            ?.map(({ properties }) => properties)
            ?.map(({ administration_id, name }) => ({
              administration_id,
              name,
            }));
          appDispatch({
            type: "SET_ADM",
            payload: administrations,
          });
        }
      }}
    />
  );
};

export default DynamicScript;
