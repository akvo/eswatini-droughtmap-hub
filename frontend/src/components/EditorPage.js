"use client";

const { useState } = require("react");
import TinyEditor from "./TinyEditor";

const EditorPage = () => {
  const [content, setContent] = useState("");

  return (
    <>
      <TinyEditor id="tiny-editor" value={content} setValue={setContent} />
      <hr />
      <div dangerouslySetInnerHTML={{ __html: content }} />
    </>
  );
};

export default EditorPage;
