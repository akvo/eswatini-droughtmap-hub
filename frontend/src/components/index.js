export * from "./Map";
export { default as TinyEditor } from "./TinyEditor";
