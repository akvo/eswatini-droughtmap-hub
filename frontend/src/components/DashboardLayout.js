"use client";

import Link from "next/link";
import { Dropdown } from "antd";
import { UserCircle } from "./Icons";
import { LogoutButton } from "./Buttons";
import { USER_ROLES } from "@/static/config";
import { getProfileDropdownItems } from "@/lib";

const DashboardLayout = ({ user, children }) => {
  const menuByRoles = getProfileDropdownItems(user);
  const menuItems = [
    ...menuByRoles,
    {
      key: 99,
      label: <LogoutButton className="dropdown-item" />,
    },
  ];
  return (
    <div className="w-full min-h-screen">
      <div className="w-full py-2 bg-primary sticky top-0 z-50">
        <div className="w-full container flex flex-row justify-between items-center">
          <div>
            <Link
              href={
                user?.role === USER_ROLES.admin ? "/publications" : "/reviews"
              }
              className="text-white"
            >
              Dashboard
            </Link>
          </div>
          <div>
            <Dropdown
              placement="bottomRight"
              menu={{
                items: menuItems,
              }}
              trigger={["click"]}
              overlayClassName="dropdown-profile"
            >
              <a
                role="button"
                className="text-white"
                onClick={(e) => e.preventDefault()}
              >
                <UserCircle size={32} />
              </a>
            </Dropdown>
          </div>
        </div>
      </div>
      <main className="w-full container">{children}</main>
    </div>
  );
};

export default DashboardLayout;
