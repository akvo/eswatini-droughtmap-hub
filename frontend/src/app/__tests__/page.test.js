import { Navbar } from "@/components";
import { render } from "@testing-library/react";

describe("HomePage", () => {
  it("renders login button when session empty", () => {
    const { getByText } = render(<Navbar />);
    expect(getByText("LOGIN")).toBeInTheDocument();
  });

  it("renders profile button when session exists", () => {
    const { getByLabelText } = render(<Navbar session={{ id: 1 }} />);

    expect(getByLabelText("Profile")).toBeInTheDocument();
  });

  it("renders correctly & match with the snapshot", () => {
    const { container } = render(<Navbar session={{ id: 1 }} />);
    expect(container).toMatchSnapshot();
  });
});
