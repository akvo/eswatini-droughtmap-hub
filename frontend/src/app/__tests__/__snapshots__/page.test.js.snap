// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage renders correctly & match with the snapshot 1`] = `
<div>
  <div
    class="w-full bg-primary"
  >
    <div
      class="w-full container flex flex-row justify-between items-center"
    >
      <nav
        class="w-full text-base"
      >
        <a
          class="text-neutral-50"
          href="/"
        >
          <button
            class="text-white px-6 py-3 transition-colors duration-300 hover:bg-rose-800"
          >
            Home
          </button>
        </a>
        <a
          class="text-neutral-50"
          href="/browse"
        >
          <button
            class="text-white px-6 py-3 transition-colors duration-300 hover:bg-rose-800"
          >
            Browse
          </button>
        </a>
        <a
          class="text-neutral-50"
          href="/compare"
        >
          <button
            class="text-white px-6 py-3 transition-colors duration-300 hover:bg-rose-800"
          >
            Compare
          </button>
        </a>
      </nav>
      <div
        class="w-fit px-4"
      >
        <a
          aria-label="Profile"
          class="ant-dropdown-trigger text-white"
          role="button"
        >
          <svg
            fill="none"
            height="32"
            viewBox="0 0 19 19"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.77246 1.89697C8.32619 1.89697 6.91239 2.32584 5.70986 3.12935C4.50732 3.93286 3.57006 5.07492 3.0166 6.4111C2.46313 7.74728 2.31832 9.21759 2.60047 10.6361C2.88263 12.0546 3.57907 13.3575 4.60175 14.3802C5.62442 15.4029 6.92738 16.0993 8.34587 16.3815C9.76435 16.6636 11.2347 16.5188 12.5708 15.9653C13.907 15.4119 15.0491 14.4746 15.8526 13.2721C16.6561 12.0695 17.085 10.6557 17.085 9.20947C17.0829 7.27071 16.3118 5.41193 14.9409 4.04102C13.57 2.6701 11.7112 1.89902 9.77246 1.89697ZM5.98121 14.0962C6.38814 13.4598 6.94872 12.936 7.61128 12.5733C8.27385 12.2105 9.01708 12.0203 9.77246 12.0203C10.5278 12.0203 11.2711 12.2105 11.9336 12.5733C12.5962 12.936 13.1568 13.4598 13.5637 14.0962C12.4797 14.9392 11.1457 15.3969 9.77246 15.3969C8.39924 15.3969 7.0652 14.9392 5.98121 14.0962ZM7.52246 8.64697C7.52246 8.20196 7.65442 7.76695 7.90166 7.39694C8.14889 7.02693 8.50029 6.73854 8.91143 6.56824C9.32256 6.39795 9.77496 6.35339 10.2114 6.44021C10.6479 6.52702 11.0488 6.74131 11.3635 7.05598C11.6781 7.37065 11.8924 7.77156 11.9792 8.20802C12.066 8.64448 12.0215 9.09688 11.8512 9.50801C11.6809 9.91914 11.3925 10.2705 11.0225 10.5178C10.6525 10.765 10.2175 10.897 9.77246 10.897C9.17573 10.897 8.60343 10.6599 8.18147 10.238C7.75952 9.81601 7.52246 9.24371 7.52246 8.64697ZM14.3962 13.3164C13.7689 12.4073 12.8867 11.704 11.8607 11.2949C12.4119 10.8609 12.814 10.2658 13.0113 9.5926C13.2086 8.91937 13.1911 8.2014 12.9614 7.53853C12.7317 6.87566 12.3012 6.30085 11.7296 5.89402C11.1581 5.4872 10.474 5.26858 9.77246 5.26858C9.07093 5.26858 8.38682 5.4872 7.81529 5.89402C7.24376 6.30085 6.8132 6.87566 6.5835 7.53853C6.3538 8.2014 6.33637 8.91937 6.53364 9.5926C6.73091 10.2658 7.13306 10.8609 7.68418 11.2949C6.6582 11.704 5.77606 12.4073 5.14871 13.3164C4.35599 12.425 3.83786 11.3234 3.65671 10.1443C3.47557 8.96518 3.63914 7.75885 4.12772 6.67056C4.6163 5.58226 5.40906 4.65841 6.41055 4.01024C7.41203 3.36207 8.57953 3.01722 9.77246 3.01722C10.9654 3.01722 12.1329 3.36207 13.1344 4.01024C14.1359 4.65841 14.9286 5.58226 15.4172 6.67056C15.9058 7.75885 16.0694 8.96518 15.8882 10.1443C15.7071 11.3234 15.1889 12.425 14.3962 13.3164Z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;
